packages/fx-core/src/plugins/resource/bot/v2/botSolution.ts

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AzureSolutionSettings, Inputs } from "@microsoft/teamsfx-api";
+import { Context } from "@microsoft/teamsfx-api/build/v2";
+import { isBotNotificationEnabled } from "../../../../common";
+import { AzureSolutionQuestionNames } from "../../../solution/fx-solution/question";
+import {
+  QuestionNames,
+  TemplateProjectsConstants,
+  TemplateProjectsScenarios,
+  TriggerTemplateScenarioMappings,
+} from "../constants";
+import { PluginActRoles } from "../enums/pluginActRoles";
+import {
+  HostTypes,
+  BotTrigger,
+  PluginBot,
+  QuestionBotScenarioToPluginActRoles,
+  CommonStrings,
+  Commands,
+} from "../resources/strings";
+import { CodeTemplateInfo } from "./interface/codeTemplateInfo";
+import * as utils from "../utils/common";
+import { HostTypeTriggerOptions } from "../question";
+import path from "path";
+import { CommandExecutionError } from "../errors";
+
+export type BicepConfigs = string[];
+
+export class BotSolution {
```

Comment:
Does this BotSolution behave as Composite component in Huajie's design? I guess it should replace the current bot plugin, which provide api v2 implementation.

Reply:
In my understanding, BotSolution should stand for bot component. Let me merge this class to bot plugin.